pkg/kv/fs.go

```diff
@@ -0,0 +1,67 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type FSStore struct {
+	lock   *sync.RWMutex
+	fsPath string
+}
+
+func NewFSStore(logger *zap.Logger, fsPath string) *FSStore {
+	return &FSStore{
+		lock:   new(sync.RWMutex),
```

Comment:
We don't need a mutex in this struct because there are no variables for us to read/write concurrently in this struct.

By default, file io in golang is thread safe.